presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -528,6 +569,7 @@ private void queryFinished(QueryExecution query)
                     group = group.parent.orElse(null);
                 }
             }
+            activeQueries.remove(query);
```

Comment:
Assuming this is to check that activeQueries also does not contain the query, added checkState.